evaluator/evaluator/app/routers/dataset.py

```diff
@@ -16,7 +26,185 @@
+    if dataset_metadata.name not in datasets:
+        message = f"Metadata entry with dataset_name={dataset_metadata.name} does not exist! Please use the POST method to create the entry."
+        logger.warning(message)
+        raise HTTPException(
+            400,
+            message,
+        )
+    result = mongo_client.client.evaluator.datasets.replace_one(
+        {"name": dataset_metadata.name}, dataset_metadata.mongo(), upsert=False
+    )
+    if result.acknowledged:
+        logger.info(
+            f"Metadata entry with dataset_name={dataset_metadata.name} successfully updated. Triggering re-download dataset task after HTTP response..."
+        )
+        background_tasks.add_task(
+            trigger_download_dataset, dataset_handler, dataset_metadata.name
+        )
+        return
+    else:
+        raise RuntimeError(result.raw_result)
+
+
+@router.delete(
+    "/metadata/{dataset_name}",
+    status_code=200,
+)
+async def delete_metadata(
+    dataset_name: str,
+    background_tasks: BackgroundTasks,
+    dataset_handler: DatasetHandler = Depends(DatasetHandler),
+    token_payload: Dict = Depends(auth),
+):
+    validate_access(token_payload)
+    logger.debug(f"Deleting metadata with name: {dataset_name}")
+
+    delete_result = mongo_client.client.evaluator.datasets.delete_one(
+        {"name": dataset_name}
+    )
+    if delete_result.deleted_count == 1:
+        logger.info(
+            f"Metadata entry with dataset_name={dataset_name} successfully deleted. Triggering delete dataset task after HTTP response..."
+        )
+        background_tasks.add_task(trigger_delete_dataset, dataset_handler, dataset_name)
+        return
+    else:
+        message = (
+            f"Dataset {dataset_name} could not be deleted, because it was not found."
+        )
+        logger.warning(message)
+        raise HTTPException(404, message)
+
+
+async def validate_mapping_and_skill_type(dataset_metadata: DatasetMetadata):
+    """
+    Function that checks if the specified skill type exists and if the given mapping has the format of the skill type.
+    """
+    if dataset_metadata.skill_type not in SUPPORTED_SKILL_TYPES.keys():
+        raise HTTPException(
+            400,
+            f'Skill type {dataset_metadata.skill_type} not supported! Please use one of the following skill types: {", ".join(SUPPORTED_SKILL_TYPES.keys())}',
+        )
+    try:
+        SUPPORTED_SKILL_TYPES[dataset_metadata.skill_type].parse_obj(
+            dataset_metadata.mapping
+        )
+    except ValidationError as error:
+        raise HTTPException(
+            400,
+            f"The mapping does not match the skill type {dataset_metadata.skill_type}: {error.errors}",
+        )
+
+
+def validate_access(token_payload: Dict):
+    """
+    Function to validate access. Currently SQuARE has no user roles, so we check for the statically defined user names.
+    """
+    ALLOWED_USER_NAMES = ["ukp", "LOCAL_SQUARE_USER"]
```

Comment:
Würde die lieber kleinschreiben und dann hier `if token_payload["username"] not in ALLOWED_USER_NAMES:` den username aus dem Token lowercasen. Nicht, dass es am Ende scheitert weil ukp groß geschrieben wird oder sowas :D

Reply:
Wollte das als extra Sicherheitsfeature, aber sicher ist das wahrscheinlich eh nicht 😂 Dann ändere ich das